backend/api/utils.py

```diff
@@ -1,70 +1,45 @@
 import json
-from bson import json_util
-from typing import Any, List, Tuple, Type
+import bson
+from typing import Any
 
 import pymongo
+
+from pymongo.collection import Collection
 from rest_framework.response import Response
 from rest_framework.request import Request
 
-from .exceptions import CollectionFull
+from django.http.request import QueryDict
+
 
+DEFAULT_SORT_KEY = "id"
 
-__Collection = type(Type[pymongo.collection.Collection])
-__ApiRequest = type(Type[Request])
+DEFAULT_SORT_ORDER = pymongo.ASCENDING
 
-MIN_ID_VALUE = 100_000_000_000
-MAX_ID_VALUE = 999_999_999_999
-DEFAULT_QUERY_LIST_SORT_KEY = "id"
-DEFAULT_QUERY_LIST_SORT_ORDER = pymongo.ASCENDING
-QUERY_LIST_SORT_ORDER = {
+SORT_ORDER = {
     "ascending": pymongo.ASCENDING,
     "descending": pymongo.DESCENDING,
 }
 
 
 def _to_python_object(obj: Any) -> Any:
-    return json.loads(json_util.dumps(obj))
+    return json.loads(bson.json_util.dumps(obj))
 
 
 class JsonResponse(Response):
     def __init__(self, data: Any, *args, **kwargs):
         super().__init__(_to_python_object(data), *args, **kwargs)
 
 
-def get_sort_order_from_request(
-    request: Request, collection: __Collection
-) -> List[Tuple[str, int]]:
-
-    sort_key = request.query_params.get("sort_key", None)
-    sort_order = request.query_params.get("sort_order", None)
-
-    if sort_key is None:
-        sort_key = DEFAULT_QUERY_LIST_SORT_KEY
-    else:
-        sort_key = str(sort_key).lower()
-        # TODO: Find a way to check if the sort_key in a valid key of the `collection`.
+def get_sort_order(query_params: QueryDict, collection: Collection):
+    sort_key = query_params.get("sort_key", DEFAULT_SORT_KEY)
+    sort_key = str(sort_key).lower()
+    # TODO: Find a way to check if the sort_key in a valid key of the `collection`.
 
-    if sort_order is None:
-        sort_order = DEFAULT_QUERY_LIST_SORT_ORDER
-    else:
-        sort_order = QUERY_LIST_SORT_ORDER.get(
-            str(sort_order).lower(), DEFAULT_QUERY_LIST_SORT_ORDER
-        )
+    sort_order = query_params.get("sort_order")
+    sort_order = SORT_ORDER.get(str(sort_order).lower(), DEFAULT_SORT_ORDER)
 
     return [(sort_key, sort_order)]
 
 
-def get_new_id(collection: __Collection) -> str:
-    if collection.find_one() is None:
-        return str(MIN_ID_VALUE)
-    else:
-        current_max_id = _to_python_object(
-            collection.find().sort("id", pymongo.DESCENDING).limit(1)
-        )[0]["id"]
-
-        if int(current_max_id) < MAX_ID_VALUE:
-            return str(int(current_max_id) + 1)
-        else:
-            raise CollectionFull(
```

Comment:
that line surprised me. We shouldn't generate the id by hand... And a full database ? Never happened to me before, so I don't think it will happen here.

The current implementation make use of `bson.ObjectId`. If you think that id of length 24 is too much, we can find alternatives solution. The generated ids will be unique since the id contain the timestamp.

Reply:
* I didn't find another way to fulfill the `12` length id requirement. As default `ObjectId` contained more characters, I searched for some `id`/`uuid` libs, but didn't find anything. Finally, I was forced to go with that solution of hand made ids. I'm completely okay with what you modified there.

* For the full database, it was just as a (useless) security, and a little for the fun of doing it. 
* About the `bson.ObjectId`:
    * Is if for space saving that you want short ids ?
    * If you find alternative solutions, we can try it. But I hope it doen't end up in truncating something, I don't feel cumfortable doing it.

Reply:
- Yeah no worries. You should have use something like `uuid4` or whatever. id of length 12 is not really a constraint. We don't care at this point. Sorry because I mislead you with length 12
- Haha
- Hum how about you come with a solution and we can discuss ? As I said, I don't care about how we generate the ids as long as we are consistent and we don't do it by hand ourselves

Reply:
Okay, I'll think about it !